tests/visualization_tests/test_pareto_front.py

```diff
@@ -1,451 +1,249 @@
+    study.enqueue_trial({"x": 1, "y": 2, "z": 1})
+    study.enqueue_trial({"x": 1, "y": 1, "z": 2})
+    study.enqueue_trial({"x": 0, "y": 2, "z": 3})
+    study.enqueue_trial({"x": 1, "y": 0, "z": 4})
+    study.optimize(
+        lambda t: [t.suggest_int("x", 0, 2), t.suggest_int("y", 0, 2), t.suggest_int("z", 0, 5)],
+        n_trials=4,
+    )
+    return study
 
-    with pytest.raises(ValueError, match=error_message):
-        plot_pareto_front(
-            study=study,
-            target_names=[],
-            include_dominated_trials=include_dominated_trials,
-            targets=targets,
-        )
 
-    with pytest.raises(ValueError, match=error_message):
-        plot_pareto_front(
-            study=study,
-            target_names=["Foo"],
-            include_dominated_trials=include_dominated_trials,
-            targets=targets,
-        )
+@pytest.mark.parametrize("include_dominated_trials", [False, True])
+@pytest.mark.parametrize("axis_order", [None, [0, 1], [1, 0]])
+@pytest.mark.parametrize("targets", [None, lambda t: (t.values[0], t.values[1])])
+@pytest.mark.parametrize("target_names", [None, ["Foo", "Bar"]])
+def test_get_pareto_front_info_unconstrained(
+    include_dominated_trials: bool,
+    axis_order: Optional[List[int]],
+    targets: Optional[Callable[[FrozenTrial], Sequence[float]]],
+    target_names: Optional[List[str]],
+) -> None:
+    if axis_order is not None and targets is not None:
+        pytest.skip("skip using both axis_order and targets")
 
-    with pytest.raises(ValueError, match=error_message):
-        plot_pareto_front(
-            study=study,
-            target_names=["Foo", "Bar", "Baz"],
-            include_dominated_trials=include_dominated_trials,
-            axis_order=axis_order,
-            targets=targets,
-        )
+    study = create_study_2d()
+    trials = study.get_trials(deepcopy=False)
 
-    target_names = ["Foo", "Bar"]
-    figure = plot_pareto_front(
+    info = _get_pareto_front_info(
         study=study,
-        target_names=target_names,
         include_dominated_trials=include_dominated_trials,
         axis_order=axis_order,
-        constraints_func=constraints_func,
         targets=targets,
+        target_names=target_names,
+    )
+
+    assert info == _ParetoFrontInfo(
+        n_targets=2,
+        target_names=target_names or ["Objective 0", "Objective 1"],
+        best_trials_with_values=[(trials[2], [0, 2]), (trials[3], [1, 0])],
+        non_best_trials_with_values=[(trials[0], [1, 2]), (trials[1], [1, 1])]
+        if include_dominated_trials
+        else [],
+        infeasible_trials_with_values=[],
+        axis_order=axis_order or [0, 1],
+        include_dominated_trials=include_dominated_trials,
+        has_constraints_func=False,
     )
-    assert figure.layout.xaxis.title.text == target_names[actual_axis_order[0]]
-    assert figure.layout.yaxis.title.text == target_names[actual_axis_order[1]]
```

Comment:
Tests for wrong number of `target_names` is moved to `test_get_pareto_front_invalid_number_of_target_names`. Also, we only need one test for such case.